tests/queries/0_stateless/03237_create_table_select_as_with_recursive.sql

```diff
@@ -0,0 +1,3 @@
+drop table if exists t;
+create view t AS (WITH RECURSIVE 42 as ttt SELECT ttt);
```

Comment:
`RECURSIVE 42 as ttt` what does it even mean? I suspect we should not allow such queries.

Reply:
yes we shouldn't allow expression as identifier in `with recursive` clause. will change the parser

Reply:
it's example only. 
real query might be more complex
but only one CTE in whole with declaration is recursive and other CTEs are regular.
for example

```sql
create view t AS (
  WITH RECURSIVE
    toDate({startDate:String}) AS _startDate
    , toDate({endDate:String}) AS _endDate
    , {preferredCategories:Array(UInt64)} AS _preferredCategories
    , {preferredSubjects:Array(Tuple(UInt64, UInt64))} AS _preferredSubjects
    , _source_categories AS (
      SELECT
        categoryId
        , argMax(dc.name, dc.date) AS _name
        , argMax(dc.path, dc.date) AS _path
        , argMax(dc.url, dc.date) AS _url
        , argMax(dc.parentId, dc.date) AS _parentId
      FROM daily_categories dc
      WHERE
        dc.date >= _startDate
        AND dc.date <= _endDate
      GROUP BY categoryId
    )
    , _low_level_categories AS (
        SELECT 
          _sc.*
          , [_sc.categoryId] AS path
          , []::Array(UInt64) AS parents
        FROM _source_categories _sc
        WHERE  
          _sc._parentId = 0
      UNION ALL
        SELECT 
          _sc.*
          , arrayConcat(_llc.path, [_sc.categoryId]) AS path
          , arrayConcat(_llc.path) AS parents
        FROM _source_categories _sc, _low_level_categories _llc
        WHERE 
          _sc._parentId = _llc.categoryId
    )
  SELECT 
    distinct(categoryId)
    , _name
    , _path
    , _url
    , _parentId
  FROM (
    SELECT 
      * 
    FROM ( 
      SELECT 
        *
        , groupUniqArrayArray(parents) OVER() AS _parents
      FROM _low_level_categories
    ) _llc
    WHERE
      _llc._parentId != 0
      AND not has(_parents, _llc.categoryId) 
      AND ((
        length(_preferredCategories) = 0 AND length(_preferredSubjects) = 0
      ) OR (
        length(_preferredCategories) != 0 AND hasAny(_llc.path, _preferredCategories)
      ))
  )
  SETTINGS
    allow_experimental_analyzer = 1
);
```

recursive needs for _low_level_categories only in that WITH declaration.

I guss that needs to test more complex queries too.

Reply:
@novikd Looks like it's a reasonable SQL, `WITH RECUSIVE (expression as alias) , (sql as table_name) ...`. What do you think?

Reply:
Yes, it makes sense. I initially thought `RECURSIVE` must be a part of the CTE declaration. I agree that it is worth adding more complex test queries.

Reply:
@novikd updated tests, PTAL